### PR TITLE
Initialization

### PR DESCRIPTION
**Pull request by ajlee21**
This PR contains the following documents:
data/  gene expression data
metadata/  label mapping files
models/  dimensionality reduction methods
scripts/  scripts to train models
stats/  tybalt log files
viz/  clustering images

### CODE REVIEW

File: scripts/nbconverted/PCA_ICA_model.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# specify either 'ica' or 'pca'
+method = 'pca' 
+
+
+# In[4]:
+
+
+# read in data
+data = pd.read_table(data_file, header = 0, sep = '\t', index_col = 0)
+data = pd.DataFrame(data)
```
Comment (danich1) [inline scripts/nbconverted/PCA_ICA_model.py]: Why do you need this line? `data` is already a dataframe after you call read table

File: scripts/nbconverted/PCA_ICA_model.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+# Note:
+# According to the source (https://github.com/scikit-learn/scikit-learn/blob/14031f6/sklearn/decomposition/pca.py#L310),
+# input will be transformed by np.array() before doing PCA. So row index will be lost during 
+# PCA.fit_transform(X) even using a structured array or a pandas DataFrame. However, the order of the data is preserved, 
+# meaning you can attach the index back afterward
+
+X_ann = pd.DataFrame(reduced_X, index=X.index, columns=['1', '2'])
+
+# read in mapping file (sample id --> phenotype)
+map = pd.read_table(map_file, header = 0, sep = '\t', index_col = 0)
```
Comment (danich1) [inline scripts/nbconverted/PCA_ICA_model.py]: map is a reserved keyword, so I would suggest a variable name change like `mapper_df` or something of that nature.

File: scripts/nbconverted/PCA_ICA_model.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+#X_new.shape
+
+
+# In[7]:
+
+
+# Plot
+fg = sns.lmplot(x = '1', y = '2', data = X_new, hue = 'medium', fit_reg = False)
+fg.add_legend()
+fig_file = 'C:/Users/alexj/Documents/UPenn/CGreene/Pseudomonas/viz/'+method+'.png'
```
Comment (danich1) [inline scripts/nbconverted/PCA_ICA_model.py]: Would be easier to write this line using string formatting: 
```python
fig_file = 'C:/Users/.../viz/{}.png'.format(method)
``` 
Getting into the habit now will make future string operations easier for you.

File: scripts/nbconverted/PCA_ICA_model.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+fg.fig.suptitle(method.upper()+' compressed data')
+fg.savefig(fig_file)
+
+
+# In[8]:
+
+
+# Output compressed data
+file_out = 'C:/Users/alexj/Documents/UPenn/CGreene/Pseudomonas/models/'+method+'_encoded.txt'
```
Comment (danich1) [inline scripts/nbconverted/PCA_ICA_model.py]: Comment above ^^

File: scripts/nbconverted/raw_model.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+data_file = 'C:/Users/alexj/Documents/UPenn/CGreene/Pseudomonas/data/all-pseudomonas-gene.pcl'
+map_file = 'C:/Users/alexj/Documents/UPenn/CGreene/Pseudomonas/metadata/mapping_sampleID_medium.txt'
+
+
+# In[4]:
+
+
+# read in data
+data = pd.read_table(data_file, header = 0, sep = '\t', index_col = 0)
+data = pd.DataFrame(data)
```
Comment (danich1) [inline scripts/nbconverted/raw_model.py]: refer to dataframe comment^^

File: scripts/nbconverted/raw_model.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+# Note:
+# According to the source (https://github.com/scikit-learn/scikit-learn/blob/14031f6/sklearn/decomposition/pca.py#L310),
+# input will be transformed by np.array() before doing PCA. So row index will be lost during 
+# PCA.fit_transform(X) even using a structured array or a pandas DataFrame. However, the order of the data is preserved, 
+# meaning you can attach the index back afterward
+X_ann = pd.DataFrame(tsne_X, index=X.index, columns=['tsne1', 'tsne2'])
+
+# read in mapping file (sample id -- phenotype)
+map = pd.read_table(map_file, header = 0, sep = '\t', index_col = 0)
```
Comment (danich1) [inline scripts/nbconverted/raw_model.py]: refer to map comment^^

File: scripts/nbconverted/tsne_viz.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+map_file = 'C:/Users/alexj/Documents/UPenn/CGreene/Pseudomonas/metadata/mapping_sampleID_medium.txt'
+
+
+# In[4]:
+
+
+# read in encoded data
+X = pd.read_table(encoded_data_file, header = 0, sep = '\t', index_col = 0)
+#X = pd.read_table(encoded_data_file, header = None, sep = '\t', index_col = None) # eADAGE doesn't have header
+X = pd.DataFrame(X)
```
Comment (danich1) [inline scripts/nbconverted/tsne_viz.py:43]: refer to dataframe comment^^

File: scripts/nbconverted/tybalt_1layer_model.py [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+epochs = 100
+kappa = 0.01
+
+original_dim = rnaseq.shape[1]
+intermediate_dim = 100
+latent_dim = 10
+epsilon_std = 1.0
+beta = K.variable(0)
+
+stat_file = 'C:/Users/alexj/Documents/UPenn/CGreene/Pseudomonas/stats/tybalt_1layer'+str(latent_dim)+'_stats.csv'
```
Comment (danich1) [inline scripts/nbconverted/tybalt_1layer_model.py]: Refer to string formatting comment above^^

**Review by danich1 — COMMENTED** [review]
Nice PR. I have some code re-design suggestions I think you should make.

File: scripts/nbconverted/PCA_ICA_model.py [hunk 9]
```diff
@@ -0,0 +1,101 @@
+
```
Comment (huqiwen0313) [inline scripts/nbconverted/PCA_ICA_model.py:1]: Can you add some description about this script ?

File: scripts/nbconverted/PCA_ICA_model.py [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+import numpy as np
+np.random.seed(123)
+
+
+# In[3]:
+
+
+# load arguments
+data_file = 'C:/Users/alexj/Documents/UPenn/CGreene/Pseudomonas/data/all-pseudomonas-gene-normalized.pcl'
+map_file = 'C:/Users/alexj/Documents/UPenn/CGreene/Pseudomonas/metadata/mapping_sampleID_medium.txt'
```
Comment (huqiwen0313) [inline scripts/nbconverted/PCA_ICA_model.py]: May be use relative path, e.g. data_file =  os.path.join("Pseudomonas", "data", "all-pseudomonas-gene-normalized.pcl")

**Review by huqiwen0313 — COMMENTED** [review]
Looks good. I have some additional minor comments

**Comment by ajlee21** [comment]
I have added updated script files based on your comments

Thank you so much!

**Review by huqiwen0313 — APPROVED** [review]
Looks good to me.
